velox/functions/prestosql/tests/DecimalArithmeticTest.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/prestosql/tests/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace facebook::velox {
+
+class DecimalArithmeticTest : public FunctionBaseTest {
+ public:
+  DecimalArithmeticTest() {
+    options_.parseDecimalAsDouble = false;
+  }
+
+ protected:
+  template <typename T>
+  void testDecimalExpr(
+      const VectorPtr& expected,
+      const std::string& expression,
+      const std::vector<VectorPtr>& input) {
+    auto result = evaluate<SimpleVector<T>>(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+    ASSERT_EQ(result->type()->toString(), expected->type()->toString());
+  }
+
+  template <typename T>
+  void testOpDictVectors(
+      const std::string& operation,
+      const VectorPtr& result,
+      const std::vector<VectorPtr>& flatVector) {
+    // Dictionary vectors as arguments.
+    vector_size_t newSize = flatVector[0]->size() * 2;
+    auto indices1 = makeIndices(newSize, [&](int row) { return row / 2; });
+    auto indices2 = makeIndices(newSize, [&](int row) { return row / 2; });
+    auto shortDictA =
+        VectorTestBase::wrapInDictionary(indices1, newSize, flatVector[0]);
+    auto shortDictB =
+        VectorTestBase::wrapInDictionary(indices2, newSize, flatVector[1]);
+    auto resultDict =
+        VectorTestBase::wrapInDictionary(indices1, newSize, result);
+
+    testDecimalExpr<T>(
+        resultDict,
+        fmt::format("{}(c0,c1)", operation),
+        {shortDictA, shortDictB});
+  }
+};
+} // namespace facebook::velox
+
+TEST_F(DecimalArithmeticTest, add) {
+  auto resultLongFlat = makeLongDecimalFlatVector({2000, 4000}, DECIMAL(19, 3));
+  auto shortFlat = makeShortDecimalFlatVector({1000, 2000}, DECIMAL(18, 3));
+  // Add short and short, returning long.
+  testDecimalExpr<LongDecimal>(
+      resultLongFlat, "plus(c0, c1)", {shortFlat, shortFlat});
+  // Add short and long, returning long.
+  auto longFlat = makeLongDecimalFlatVector({1000, 2000}, DECIMAL(19, 3));
+  resultLongFlat = makeLongDecimalFlatVector({2000, 4000}, DECIMAL(20, 3));
```

Comment:
nit: `expectedResult` should do.

Reply:
I suffix with long and short because the flat vectors will be of different types. Cannot reuse same variable for long and short. Will add prefix expected.